### DISCUSSION TITLE
Ironbar discord or matrix

### DISCUSSION THREAD

**Discussion by yavko** [JakeStanger/ironbar#126] [Ideas]
I think it would be cool if there was a discord or matrix, so us ironbar users could discuss stuff about ironbar in real-time

**Comment by JakeStanger** [comment]
I appreciate the interest, but I've a few concerns before jumping into it.

So my main question would be this - what would you get out of it that you don't from GitHub? 

My problem with software Discords (or Matrices (is that the right plural?)) is that discussion that should really be taking place on GitHub inside issue threads and discussions ends up happening in the server (I'd want to retain help threads and show-and-tells to the discussions here). 

The consequence of that is that it becomes much harder to find resources by googling, and instead you have to join the discord and hope what you're looking for comes up in the search engine.

I'm also not sure if Ironbar has a big enough community to warrant a server yet, but I guess we'll see - anybody interested, let me know here :p

If there is enough interest for a server, and enough reason to set one up, I'll do it.

**Reply by yavko** [reply]
For bugs and such you are definitely right, but if that's the case the server/space could be restricted to discussions & development, and people asking for help can be redirected to gh discussions for help, and gh issues for bugs

**Reply by JakeStanger** [reply]
Development discussion I'd like to keep on GH too, since I want all discussions to be fully accessible/open to everyone (to at least read).

It may be that Ironbar is also too niche for a full blown server, instead being better suited to a single channel (perhaps a suitable server already exists). I think if/when the community grows a bit, if this has interest, we can review the situation and look into setting something up for discussions.

**Reply by HenrySteinmetz** [reply]
I think the only thing not having a discord does is moving discussions to other discords (e.g. hyprland development server) 
I would really appreciate  a place to discuss errors and share configs more open than the github discussions tab that literally no one knows about.

**Reply by yavko** [reply]
> I think the only thing not having a discord does is moving discussions to other discords (e.g. hyprland development server) 
> I would really appreciate  a place to discuss errors and share configs more open than the github discussions tab that literally no one knows about. 

Yeah I agree on this, I often see people sharing ironbar stuff for hypr development, and only a couple people use discussions anyway

**Reply by yavko** [reply]
> > I think the only thing not having a discord does is moving discussions to other discords (e.g. hyprland development server) 
> > I would really appreciate  a place to discuss errors and share configs more open than the github discussions tab that literally no one knows about. 

> Yeah I agree on this, I often see people sharing ironbar stuff (mostly people asking questions if it's good) in hypr development, and only a couple people use discussions anyway

**Reply by JakeStanger** [reply]
> more open than the github discussions tab that literally no one knows about

I have a couple of issues with this statement. 

Firstly, requiring users to sign up to a service and join a server just to *read* discussions is not more open than having them publicly available and findable on search engines. 

The idea that "literally nobody" knows about GH discussions is clearly false given you found your way to this discussion. I'll agree that discussions are quieter, but I don't think that detracts from the point that they're the right place to be holding a lot of discussions. Imagine you're a new user, you google an error, and nothing comes up, because a relevant discussion has taken place on the non-accessible Discord server. That's frustrating.

> I think the only thing not having a discord does is moving discussions to other discords

I fully get this point. The issue is there will always be some degree of splintering (people will always talk about things across servers). This comes down to having a large enough user base that the server would hold at least most of the conversation, rather than just splitting it further.

I also do think there would be some merit in it for the sake of reaching a larger audience regarding breaking changes.

Again, this isn't a hard no. I think I have a few criteria for it:

1) There would have to be a bot that published relevant content from the server back to a publicly accessible place, either in GitHub or elsewhere (like IRC logs) so that people can find issues without needing to join/search the server.
2) There would have to be a hard distinction between what goes on Discord and what goes on GitHub. Currently that's hard to define and I don't have the time to enforce it or really set it up.
3) Related to the last part of point 2, there would need to be a large enough community. It's hard to tell if Ironbar is at the point, but if it's too small it risks dying.

Please do continue to add any arguments/reasons you have because I do want to hear them, and also if you don't have any comments to make please just upvote the original post, because it helps show interest. 

It sounds like in the meantime, there are communities that you are able to talk about Ironbar, without the need to have a full dedicated one. It's also very cool to hear there are discussions happening!

**Reply by yavko** [reply]
For now, I think it may be better to add something to the README pointing users over to the discussions page, but personally I think excluding all the purposes of issues, discussions, and perhaps PRs, would be just a place to connect with other ironbar users, as GH is a more professional feeling place, discord/matrix are more casual and idle. And to do with the problem with people using discord/matrix instead of GH issues/discussions/PRs, a clause could be added to the rules, and users could instead point users to the respective pages. Along with that a help channel/room could be made, which instead of being interactable could tell/lead users to instead ask for help on the GH discussions, and same thing for issues too, but for bugs/feature requests, though this one is a bit extreme, could really help lead users away from the discord/matrix space instead of the GH respective pages. Thanks for being open to our ideas!

**Reply by yavko** [reply]
Also, to elaborate on the README part, I though something like maybe giving the idea of users to post their configs in show and tell. Or maybe even inviting people to look at show and tell in the `## Styling` section
Ex.
```md
✨ Have a nice config? Try showing it off in [Show and Tell](https://github.com/JakeStanger/ironbar/discussions/categories/show-and-tell)
```
Though this can also be applied differently, this is just an example :)

**Reply by JakeStanger** [reply]
These all seem like solid points and ideas. The README I'll definitely get updated; I might rework it a bit in the process just to keep it fresh/up to date.

The one thing I would say is that it's all well and good writing server rules but people do love to ignore them (getting people to use the bug issue template to log bugs here is difficult enough :p). That again comes back to time - I think in a couple of months, I should have some more I can dedicate to this, and by then hopefully the community will have grown a bit, and we can revisit then,
